docker/unified-builder

```diff
@@ -126,29 +128,40 @@ RUN --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked \
   && mv -f target/${CARGO_BUILD_TARGET}/release/opa-tp /artifacts/${TARGETARCH}
 
 
-FROM --platform=${BUILDPLATFORM} crossbuild AS test
+FROM crossbuild AS testbase
+ARG DOCKER_GID
+
+USER root
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - \
+  && echo "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
+  && apt-get update -y \
+  && apt-get install -y --no-install-recommends \
+  docker-ce-cli
+
+RUN groupadd -g ${DOCKER_GID} docker \
+  && usermod -aG docker tester
+
+
+FROM --platform=${BUILDPLATFORM} testbase AS test
 ARG BUILDPLATFORM
-ARG BUILD_ARGS
 
 # Test on the host only
 # PostgreSQL will not run as root
 WORKDIR /app
 
-USER root
-RUN --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked \
-  chown -R tester /usr/local/cargo/registry
-
 USER tester
-RUN --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked \
-  if [ "$BUILDPLATFORM" = "linux/arm64" ]; then \
+RUN if [ "$BUILDPLATFORM" = "linux/arm64" ]; then \
   export CARGO_BUILD_TARGET=aarch64-unknown-linux-gnu; \
   elif [ "$BUILDPLATFORM" = "linux/amd64" ]; then \
   export CARGO_BUILD_TARGET=x86_64-unknown-linux-gnu; \
   else \
   echo "Unsupported architecture: $(BUILDPLATFORM)"; \
   exit 1; \
   fi \
-  && cargo test --locked --release --target ${CARGO_BUILD_TARGET}
+  && echo ${CARGO_BUILD_TARGET} >cargo-build-target
```

Comment:
This is weird.  In the test, you aren´t going to have to do --target at all, since it will always be the buildplatform.